Add system and libraries information to render log

Mango-3 opened the pull request:
- Adds detailed information in the render log about the system as well as the appleseed and 3rd. party libraries linked with the plugin (feature request #313).

The screenshot below shows a test output of the render log with the added information:
![Capture](https://user-images.githubusercontent.com/22252558/66669103-72961f00-ec5f-11e9-80ba-8b3bd0cd5eaf.PNG)

dictoon commented on src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp:
```diff
@@ -2157,7 +2155,14 @@ int AppleseedRenderer::Render(
+        print_third_party_libraries_information();
+        asf::System::print_information(asr::global_logger());
+    }
+
```
Keep a single blank line.

dictoon commented on src/appleseed-max-impl/utilities.h:
```diff
@@ -74,7 +74,6 @@ class Texmap;
 
 void update_map_buttons(IParamMap2* param_map);
 
-
```
Revert, we use two blank lines before block comments.

dictoon commented on src/appleseed-max-impl/utilities.h:
```diff
@@ -129,6 +128,17 @@ std::string insert_procedural_texture_and_instance(
+// Version information functions.
+//
+
+extern void print_libraries_information();
```
The `extern` qualifiers are redundant, you can remove them.

dictoon commented on src/appleseed-max-impl/utilities.h:
```diff
@@ -129,6 +128,17 @@ std::string insert_procedural_texture_and_instance(
     renderer::ParamArray        texture_params = renderer::ParamArray(),
     renderer::ParamArray        texture_instance_params = renderer::ParamArray());
 
```
Add additional blank line.

dictoon commented on src/appleseed-max-impl/utilities.h:
```diff
@@ -129,6 +128,17 @@ std::string insert_procedural_texture_and_instance(
+extern const char* to_enabled_disabled(const bool value);
+
+extern void print_third_party_libraries_information();
 
```
Add additional blank line.

dictoon commented on src/appleseed-max-impl/utilities.cpp:
```diff
@@ -68,6 +70,89 @@
+void print_libraries_features()
+{
+        const bool WithDisneyMaterial =
+    #ifdef APPLESEED_WITH_DISNEY_MATERIAL
```
Left-align preprocessor statements.

dictoon commented on src/appleseed-max-impl/utilities.cpp:
```diff
@@ -68,6 +70,89 @@
+    return value ? "enabled" : "disabled";
+}
+
+void print_libraries_information()
```
`print_appleseed_library_information()` would be clearer.

dictoon commented on src/appleseed-max-impl/utilities.cpp:
```diff
@@ -68,6 +70,89 @@
+        asf::Compiler::get_compiler_version());
+}
+
+void print_libraries_features()
```
Similarly, `print_appleseed_library_features()` here.

dictoon commented on src/appleseed-max-impl/utilities.cpp:
```diff
@@ -68,6 +70,89 @@
+        const asf::APIStringPair& version = versions[i];
+        const char* lib_name = version.m_first.c_str();
+        const char* lib_version = version.m_second.c_str();
+        const size_t lib_name_length = strlen(lib_name);
```
`std::strlen()`

(And yes, the `std::` qualifier is missing on appleseed's side.)

Mango-3 commented:
Addressed your comments. True code exposed from appleseed.shared would be available to all plugins and is the better option long-term. This can be refactored at a later point.

dictoon commented:
Follow-up tickets:
- appleseed: https://github.com/appleseedhq/appleseed/issues/2722
- appleseed-max: https://github.com/appleseedhq/appleseed-max/issues/329
